enterprise/app/trends/drilldown_page.tsx

```diff
@@ -0,0 +1,465 @@
+
+  renderBucketName(v: number) {
+    if (isExecutionMetric(this.selectedMetric.metric)) {
+      return "execution" + (v === 1 ? "" : "s");
+    } else {
+      return "invocation" + (v === 1 ? "" : "s");
+    }
+  }
+
+  renderYBucketValue(v: number): string {
+    if (isExecutionMetric(this.selectedMetric.metric)) {
+      switch (this.selectedMetric.metric.execution) {
+        case stat_filter.ExecutionMetricType.QUEUE_TIME_USEC_EXECUTION_METRIC:
+          return (v / 1000000).toFixed(2) + "s";
+        default:
+          return v.toString();
+      }
+    } else {
+      switch (this.selectedMetric.metric.invocation) {
+        case stat_filter.InvocationMetricType.DURATION_USEC_INVOCATION_METRIC:
+          return (v / 1000000).toFixed(2) + "s";
+        case stat_filter.InvocationMetricType.CAS_CACHE_MISSES_INVOCATION_METRIC:
+        default:
+          return v.toString();
+      }
+    }
+  }
+
+  toStatFilterList(s: HeatmapSelection): stat_filter.StatFilter[] {
+    const updatedAtUsecMetric = isExecutionMetric(this.selectedMetric.metric)
+      ? stat_filter.Metric.create({ execution: stat_filter.ExecutionMetricType.UPDATED_AT_USEC_EXECUTION_METRIC })
+      : stat_filter.Metric.create({ invocation: stat_filter.InvocationMetricType.UPDATED_AT_USEC_INVOCATION_METRIC });
+    return [
+      stat_filter.StatFilter.create({
+        metric: updatedAtUsecMetric,
+        min: Long.fromNumber(s.dateRangeMicros.startInclusive),
+        max: Long.fromNumber(s.dateRangeMicros.endExclusive - 1),
+      }),
+      stat_filter.StatFilter.create({
+        metric: this.selectedMetric.metric,
+        min: Long.fromNumber(s.bucketRange.startInclusive),
+        max: Long.fromNumber(s.bucketRange.endExclusive - 1),
+      }),
+    ];
+  }
+
+  fetchDrilldowns() {
+    if (!this.currentHeatmapSelection) {
+      this.setState({ drilldownData: undefined });
+      return;
+    }
+    this.setState({ loadingDrilldowns: true, drilldownsFailed: false });
+    const filterParams = getProtoFilterParams(this.props.search);
+    const drilldownRequest = stats.GetStatDrilldownRequest.create({});
+    drilldownRequest.query = new stats.TrendQuery({
+      host: filterParams.host,
+      user: filterParams.user,
+      repoUrl: filterParams.repo,
+      branchName: filterParams.branch,
+      commitSha: filterParams.commit,
+      command: filterParams.command,
+
+      role: filterParams.role,
+      updatedBefore: filterParams.updatedBefore,
+      updatedAfter: filterParams.updatedAfter,
+      status: filterParams.status,
+    });
+    drilldownRequest.filter = this.toStatFilterList(this.currentHeatmapSelection);
+    drilldownRequest.drilldownMetric = this.selectedMetric.metric;
+    rpcService.service
+      .getStatDrilldown(drilldownRequest)
+      .then((untypedResponse) => {
+        const response = untypedResponse as stats.GetStatDrilldownResponse;
+        this.setState({ drilldownData: response });
+      })
```

Comment:
My IDE seems to be really finicky about whether or not it feels like resolving these types, and I have no idea why.  Removed all of these.

Reply:
FWIW, I think in the past I have run into similar issues and it's been because the proto sources/sgp stuff is somehow out of date and there's a red squiggly in the proto import. Maybe you are hitting a different issue though :shrug: